langchain/src/tools/gmail/get_thread.ts

```diff
@@ -0,0 +1,103 @@
+import { google } from "googleapis";
+import { CallbackManagerForToolRun } from "../../callbacks/manager.js";
+import { GmailBaseTool, GmailBaseToolParams } from "./base.js";
+
+export class GmailGetThread extends GmailBaseTool {
+    name = "gmail_get_thread";
+
+    description = "Get a thread from Gmail";
+
+    constructor(fields: GmailBaseToolParams) {
+        super(fields);
+      }
+    
+    async _call(threadId: string, runManager?: CallbackManagerForToolRun) {
+        const auth = await this.getAuth();
+    
+        const gmail = google.gmail({ version: "v1", auth });
+    
+        const thread = await gmail.users.threads.get({
+        userId: "me",
+        id: threadId,
+        });
+    
+        const { data } = thread;
+    
+        if (!data) {
+        throw new Error("No data returned from Gmail");
+        }
+    
+        const { messages } = data;
+    
+        if (!messages) {
+        throw new Error("No messages returned from Gmail");
+        }
+
+        const messageDataList = [];
```

Comment:
nit: can we replace this variable entirely by using functional code? So instead of looping over all `messages` we can just return `messages.map((message) => {.....})` right